Production/python/customiseHLT.py

```diff
@@ -0,0 +1,147 @@
+      vy = Var("vy", float, doc='y coordinate of vertex position'),
+      vz = Var("vz", float, doc='z coordinate of vertex position'),
+      vertexChi2 = Var("vertexChi2", float, doc='chi-squares'),
+      vertexNdof = Var("vertexNdof", float, doc='Number of degrees of freedom,  Meant to be Double32_t for soft-assignment fitters'),
+      pdgId = Var("pdgId", int, doc='PDG identifier'),
+      status = Var("status", int, doc='status word'),
+      longLived = Var("longLived", bool, doc='is long lived?'),
+      massConstraint = Var("massConstraint", bool, doc='do mass constraint?'),
+      # source: cmssw/PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc 
+      CKFtrackDz = Var("? trackRef.isNonnull && trackRef.isAvailable ? trackRef.dz : -999.", float, doc = "CKF track Dz"),
+      GSFtrackDz = Var("? gsfTrackRef.isNonnull && gsfTrackRef.isAvailable ? gsfTrackRef.dz : -999.", float, doc = "GSF track Dz"),
+      CKFtrackDxy = Var("? trackRef.isNonnull && trackRef.isAvailable ? trackRef.dxy : -999.", float, doc = "CKF track Dxy"),
+      GSFtrackDxy = Var("? gsfTrackRef.isNonnull && gsfTrackRef.isAvailable ? gsfTrackRef.dxy : -999.", float, doc = "GSF track Dxy"),
+      CKFtrackIsValid = Var("trackRef.isNonnull && trackRef.isAvailable", bool, doc = "CKF track is valid"),
+      GSFtrackIsValid = Var("gsfTrackRef.isNonnull && gsfTrackRef.isAvailable", bool, doc = "GSF track is valid"),
+      
+      # CKFLostInnerHits = Var("? trackRef.isNonnull && trackRef.isAvailable ? trackRef.hitPattern.numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS) : -999.", float, doc = "CKF track LostInnerHit"), 
+      # This doesn't work: failed to parse "? trackRef.isNonnull && trackRef.isAvailable ? trackRef.hitPattern.numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS) : -999."
+      # also tried: CKFLostInnerHits = Var("? trackRef.isNonnull && trackRef.isAvailable ? trackRef.hitPattern.numberOfLostHits : -999.", float, doc = "CKF track LostInnerHit"), 
+      # this doesn't work: method named "numberOfLostHits" for type "reco::HitPattern" was passed the wrong number of arguments. 
+      rawHcalEnergy = Var("rawHcalEnergy", float, doc='rawHcalEnergy'),
+      rawEcalEnergy = Var("rawEcalEnergy", float, doc='rawEcalEnergy'),
+
+    )
+  )
+
+  process.tauTablesTask = cms.Task(process.tauTable)
+  process.pfCandTablesTask = cms.Task(process.pfCandTable)
+  process.nanoTableTaskFS = cms.Task(process.genParticleTablesTask, process.genParticleTask,
+                                     process.tauTablesTask, process.pfCandTablesTask)
+  process.nanoSequenceMC = cms.Sequence(process.nanoTableTaskFS)
+  process.finalGenParticles.src = cms.InputTag("genParticles")
+
+
+  process.MessageLogger.cerr.FwkReport.reportEvery = 100
+  process = customiseGenParticles(process)
+
+  process.genParticleTable.variables.vx = Var('vertex().x', float, precision=10, doc='x coordinate of the gen particle production vertex')
```

Comment:
It looks like you already fill these genvertex variables in `customiseGenParticles()`, so these lines can be removed as duplicates.